### PR TITLE
hopefully fix CreateNPCTrainer test failing

### PR DESCRIPTION
**Pull request by DizzyEggg**
This time for real tho

### CODE REVIEW

File: src/battle_main.c [hunk 1]
```diff
@@ -1984,7 +1984,10 @@ u8 CreateNPCTrainerPartyFromTrainer(struct Pokemon *party, const struct Trainer
             else if (partyData[i].gender == TRAINER_MON_FEMALE)
                 personalityValue = (personalityValue & 0xFFFFFF00) | GeneratePersonalityForGender(MON_FEMALE, partyData[i].species);
             if (partyData[i].nature != 0)
-                ModifyPersonalityForNature(&personalityValue, partyData[i].nature - 1);
+            {
+                while (GetNatureFromPersonality(personalityValue) != partyData[i].nature - 1)
+                    ModifyPersonalityForNature(&personalityValue, partyData[i].nature - 1);
```
Comment (AsparagusEduardo) [inline src/battle_main.c:1989]: I worry about this `while`, and I'm getting different amount of rerolls based on code changes
image

Comment (DizzyEggg) [reply src/battle_main.c:1989]: I think the function modifying nature isn't working like it should and that's why it was failing before.

Comment (AsparagusEduardo) [reply src/battle_main.c:1989]: Then we should fix that instead of doing a `while` I feel ^^;

**Comment by AsparagusEduardo** [comment]
uuuh, how is this even compiling without modern?
```
#if MODERN
#define abs(x) (((x) < 0) ? -(x) : (x))
#endif
```
There's `abs(x)` in tons of places

**Comment by mrgriffin** [comment]
> uuuh, how is this even compiling without modern?

iirc agbcc defines `abs` even though modern doesn't. I know, I know, it's a travesty. I had assumed it was the one coming from [`stdlib.h`](<https://en.cppreference.com/w/c/numeric/math/abs>), but that doesn't really make sense because if so why isn't it available in modern? So perhaps it's some builtin? idk :sweat_smile: 

fwiw, I agree with your suggestion that it should be possible to make this work without a `while` Edu. (Or if not, then there's probably at least a better way to structure the code)

**Pull request closed without merging**